aten/src/ATen/native/quantized/cpu/qnnpack/src/q8dwconv/up8x9-aarch32-neon-per-channel.S

```diff
@@ -0,0 +1,464 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include <qnnpack/assembly.h>
+#include <requantization/runtime-assembly.h>
+
+.syntax unified
+
+# void pytorch_q8dwconv_ukernel_up8x9_per_channel__aarch32_neon(
+#     size_t channels,
+#     size_t output_width,
+#     const uint8_t** input,
+#     const void* weights,
+#     uint8_t* output,
+#     size_t input_stride,
+#     size_t output_increment,
+#     const union pytorch_qnnp_conv_quantization_params quantization_params[restrict static 1])
+BEGIN_FUNCTION pytorch_q8dwconv_ukernel_up8x9_per_channel__aarch32_neon
+    .arm
+#ifndef __APPLE__
+    .arch armv7-a
+    .fpu neon
+#endif
+
+    # Load params
+    # - r12 = quantization_params
+    LDR r12, [sp, 12]
+
+    PUSH {r4, r5, r6, r7, r8, r9, r10, r11, lr}
+    VPUSH {d8-d15}
+
+    STR r0, [sp, #-8]
+    STR r3, [sp, #-4]
+    STR r1, [sp, #-12]
+    STR r2, [sp, #-16]
```

Comment:
So you are pushing these four registers on the stack, right?  Is there a reason you are not using PUSH?

Reply:
Mainly for this reason: There are other places in the code that do not do push/pops for stack manipulation. Part of that code and part of my modifications only really do read-only access to some of those stack variables, so you dont really want to pop them out.